### DISCUSSION TITLE
Quitar ruido suelo 2º Cobertura

### DISCUSSION THREAD

**Discussion by manudona92** [manudona92/MapantES#34] [General]
He estado recapacitando sobre como quitar el ruido de la forma más eficiente, con las menos herramientas posibles. 

¿Por qué? Porque estoy viendo que hay lotes de la 1º cobertura siguen teniendo ruido
Creo que una buena solución con LASTOOLS es añadiendo la herramienta _lasnoise_

Este paso para una cuadricula de 2x2 km de 30 MB ha tardado 8''. Se implementaría cuando acabe de recortar y hacer el buffer de la cuadrícula final

lasnoise -i "D:\Software\LAStools\bin\PNOA_2016_MUR-VAL-CLM_616-4230_ORT-CLA-RGB.laz" -ignore_class 3 4 5 6 7 12 -step_xy 2 -step_z 1 -remove_noise -olaz

Este paso sería implementarlo en MapAnt Client solo para aquellas cuadrículas que necesiten reclasificación de la 2º Cobertura. 
Os dejo solo unas pruebas que hice con la Sierra del Molino: 

![image](https://user-images.githubusercontent.com/76484741/107144772-acebfb00-693d-11eb-9f58-c5b8da10c699.png)

![image](https://user-images.githubusercontent.com/76484741/107144767-a3fb2980-693d-11eb-8c77-1bbedc6df107.png)

**Comment by agustincaballero** [comment]
Lo que pasa es que ha quitado todos los cortados, ¿no? Lo que hemos procesado con la primera cobertura sí que da los cortados reales:

![image](https://user-images.githubusercontent.com/18677366/107145784-30a8e600-6944-11eb-9dfe-4371c73f2ffd.png)

Por otra parte, habría en todo caso que pedir permiso al autor, porque las2noise no está entre las que son totalmente gratuitas.

**Reply by manudona92** [reply]
Hola, 

Adjunto la parte norte de esa captura de pantalla anterior tuya, de la 2º cobertura sin ruido. 

![pullautus_depr](https://user-images.githubusercontent.com/76484741/107150024-06fbb900-695c-11eb-97d9-b9fa081a7b62.png)

Quita algún cortado de más, pero creo que no es visible a primera vista. 
Mi visión sería, donde perdemos más en la 1º cobertura y la 2º cobertura tenga ruido de suelo, utilizar esta herramienta lasnoise para la 2º cobertura y olvidarnos de la primera. Por ejemplo, para Alicante, Murcia y Madrid, a mi me gusta más la 1º (ya que en Madrid está mejor clasificada la piedra). Pero por ejemplo para Aragón Sur de la primera cobertura, CYL SE , Valencia o Castellón, utilizaría la 2º cobertura con esta herramienta de ruido. 

PS- estoy esperando a ver que dice Martin Isengburg de lastools a lo de lasnoise, ya le envie correo. También le envie a Jarkko sobre si es posible quitar el ruido desde su propio programa.

**Reply by manudona92** [reply]
Perdona, añade en -ignore class la clase 1 (unclassified)

lasnoise -i "D:\Software\LAStools\bin\PNOA_2016_MUR-VAL-CLM_616-4230_ORT-CLA-RGB.laz" -ignore_class 1 3 4 5 6 7 12 -step_xy 2 -step_z 1 -remove_noise -olaz

**Comment by agustincaballero** [comment]
Sobre lo del lasnoise: creo que en bastantes casos se habrá sobrepasado el límite de 5 millones de puntos para que no introduzca distorsión.

Mirando en https://mapant.es/Estadisticas.aspx , en la columna MB hay bastantes casos en los que se superan los 25-30 megas, que entiendo que serán los que acaben teniendo más de 5 millones de puntos.

No sé si todos esos casos corresponden a sitios donde la densidad de LAZ sea mayor.

En cualquier caso, pienso que el cliente deberá analizar siempre el fichero de entrada, y no procesar nunca si se superan los 5 millones de puntos, ¿no?

**Reply by manudona92** [reply]
Creo que podemos hacer una prueba final con lo que hay de la 2º Cobertura de Calahorra (La Rioja, actualmente procesado con 1º cobertura), que están en 2x2 km. Esta se re procesaría con Madrid_2_ruido.ini. 

Cada tile de La Rioja, 2x2 km, pesa aprox >120 mb y tiene >10M de puntos por tile. Si lo dejas disponible para procesar, lo puedo dejar procesando esta tarde y ver que tipo de distorsión. Así salimos de dudas con lo más "radical", aunque el dia de mañana esté todo en 1x1 km o menos. 

Por otro lado, viendo los fichero LAZ de >50mb no he visto cambios muy significativos, solo lo que me esperaba= reduce un poco el verde, pero creo que es por el .INI que he metido y no es un cambio significativo. 

Si te parece, hacemos esa prueba y evaluamos.

**Comment by manudona92** [comment]
Si veo algo diferente, en el de ruido, es que los bordes tienden a ser más negro de cortado, pero no sé si esto entra dentro del "buffer" que luego se corta (¿?)
![image](https://user-images.githubusercontent.com/76484741/107245401-67a7f600-6a2f-11eb-9f9c-0d2a4b9976d9.png)

**Reply by agustincaballero** [reply]
Sí, esa parte se recorta luego (unos 50/0.4233333=118 pixels)

**Comment by manudona92** [comment]
Lo que más nos debe de preocupar del mensaje es el de "tiny XYZ noise" y "points permuted"

Por ejemplo, cogiendo una cuadrícula de la Rioja 2º Cobertura, que le he aplicado lasnoise y tenía >5M de puntos, veo lo siguiente: 

Permuta los puntos en X e Y alrededor de 10-15 cm por punto. En Z 1-2 cm

Quita algunos puntos de vegetación media, muy pocos. 

Creo que lo mejor es verlo con algún caso más radical como el de la Rioja en 2x2 km para asegurarnos que todas las cuadrículas van a a tener >5M puntos.

**Comment by agustincaballero** [comment]
Hola, he observado que en la zona de Sierra del Molino, la mayoría de cuadrículas están bastante bien después de aplicar lo del ruido, pero la del noroeste está bastante mal (GID 376164231):

![image](https://user-images.githubusercontent.com/18677366/107412725-38b58100-6b10-11eb-8c50-6921780cbfe3.png)

Esa zona no tiene esas piedras; en especial, lo que te señalo con la flecha es una zona de bosque blanco limpio.

**Reply by agustincaballero** [reply]
Es curioso, pero dos cuadrículas procesadas anoche casi seguidas dan resultados bastante diferentes:

![image](https://user-images.githubusercontent.com/18677366/107413311-e9238500-6b10-11eb-80f5-a05a90e87bf5.png)

Parece que puede venir de que la de la derecha se procesó con Manu2B y la de la izquierda con ACB (el mío). ¿Versiones diferentes de lasnoise? Es lo único que se me ocurre
